graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java

```diff
@@ -267,47 +296,39 @@ public JsonNode rawClusterStats() {
 
     @Override
     public Map<String, NodeInfo> nodesInfo() {
-        final Request request = new Request("GET", "/_nodes");
-        final JsonNode nodesJson = jsonApi.perform(request, "Couldn't read Opensearch nodes data!");
-
-        return toStream(nodesJson.at("/nodes").fields())
-                .collect(Collectors.toMap(Map.Entry::getKey, o -> createNodeInfo(o.getValue())));
+        final var response = client.execute(c -> c.nodes().info(), "Could not read nodes list from indexer!");
+        return response.nodes().entrySet()
+                .stream().collect(Collectors.toMap(Map.Entry::getKey, entry -> createNodeInfo(entry.getValue())));
     }
 
-    private NodeInfo createNodeInfo(JsonNode nodesJson) {
+    private NodeInfo createNodeInfo(org.opensearch.client.opensearch.nodes.info.NodeInfo nodeInfo) {
         return NodeInfo.builder()
-                .version(nodesJson.at("/version").asText())
-                .os(nodesJson.at("/os"))
-                .roles(toStream(nodesJson.at("/roles").elements()).map(JsonNode::asText).toList())
-                .jvmMemHeapMaxInBytes(nodesJson.at("/jvm/mem/heap_max_in_bytes").asLong())
+                .version(nodeInfo.version())
+                .os(nodeInfo.os())
```

Comment:
This is a problem. 
The telemetry cluster resource and also archiving relied on this being a JsonNode.
Currently the telemetry does not contain any os info anymore:

```
    "nodes": {
      "imEvjtzgS7ytSLMpnGHmKQ": {
        "version": "2.11.0",
        "os": {},
        "jvm_mem_heap_max_in_bytes": 2147483648,
        "roles": [
          "ClusterManager",
          "Data",
          "Ingest",
          "RemoteClusterClient",
          "Search"
        ]
      },
```

Reply:
in 5.2 this looks like
```
    "nodes": {
      "lB9tkcD_QWmJEf27s3YiCA": {
        "version": "1.3.13",
        "os": {
          "refresh_interval_in_millis": 1000,
          "name": "Linux",
          "pretty_name": "Amazon Linux 2",
          "arch": "amd64",
          "version": "5.15.0-1045-aws",
          "available_processors": 2,
          "allocated_processors": 2
        },
```

Reply:
Refs #18652.